device/protocol_mqtt_v1.go

```diff
@@ -237,11 +237,6 @@ func astarteOnConnectHandler(d *Device, client mqtt.Client) {
 		}
 	}
 
-	// If a device connected for the first time, since we do not ask
-	// for a clean session and do not change its clientID, we can assume
-	// that after connection a session is present
-	d.sessionPresent = true
```

Comment:
Actually, I think we have to switch this back to false here (or inside the `if` to be clearer, probably).
The only place where where `d.sessionPresent` would currently be set with this MR is the initial connection. This means that we would just hold onto the first setting forever (even for reconnections), while the flag has to be reset each time around (and will be possibly be set to `true` in the future by the internal reconnection mechanism).